src/core/libmaven/groupFiltering.cpp

```diff
@@ -98,12 +103,14 @@ bool GroupFiltering::filterByMS1(PeakGroup &peakgroup)
 
 bool GroupFiltering::filterByMS2(PeakGroup& peakgroup)
 {
+    if (peakgroup.ms2EventCount == 0)
```

Comment:
I would expect `filterByMS2` to return `true` if it was not successful in matching the pattern, regardless of the presence/absence of fragmentation data. thoughts?

Reply:
As a standalone function in itself, yes. I agree. I will change the logic and the calls to the function accordingly.